### PR TITLE
Skills

### PR DESCRIPTION
**Pull request by hys25**

### CODE REVIEW

**Comment by hys25** [comment]
Hi @ArthurGorbenko, please check when you'll have time

File: src/App.js [hunk 1]
```diff
@@ -18,6 +19,7 @@ function App() {
         <Route path="/new-project" element={<ProjectCreate />} />
         <Route path="/project/:id" element={<ProjectView />} />
         <Route path="/project/edit/:id" element={<ProjectEdit />} />
+        <Route path="/skills" element={<AddSkill />} />
```
Comment (ArturHorbenko) [inline src/App.js]: Try to be more consistent when you pick names for routes. So if we already go with a singular `/project` route, let's do the same for skill entity.

Comment (ArturHorbenko) [reply src/App.js]: Same for folder name, `skills` -> `skill`

File: src/config/store.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { configureStore } from "@reduxjs/toolkit"
 import authReducer from "../features/auth/authSlice"
 import projectReducer from "../features/project/projectSlice"
+import skillsReducer from "../features/skills/skillsSlice"
 
 export const store = configureStore({
   reducer: {
     auth: authReducer,
     project: projectReducer,
+    skills: skillsReducer,
```
Comment (ArturHorbenko) [inline src/config/store.js]: Same for reducer name

File: src/elements/Input.js [hunk 3]
```diff
@@ -6,14 +6,17 @@ export function StyledInput({
...
   ...props
 }) {
   return (
     <div
       className={`flex flex-col relative items-start text-black mb-[10px] ${classNameWrapper}`}
     >
       <input
-        className={`text-white px-5 w-full h-[60px] text-14 appearance-none focus:outline-none bg-greyDark placeholder:text-grey ${className}`}
+        className={`text-white w-full text-14 appearance-none focus:outline-none bg-greyDark placeholder:text-grey ${
+          size ? "h-[30px] px-4" : "h-[60px] px-5"
```
Comment (ArturHorbenko) [inline src/elements/Input.js]: Better have an object to group styles by some specific name
For example : 
`{
    small : 'h-[30px] px-4',
    default: 'h-[60px] px-5',
  }`
  And then pass corresponding prop to the element. You can set `size` to have a default value ,`default` for example. 
  Going this way we'll have more scalable solution.

File: src/layout/Skill.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  deleteSkill,
+} from "../features/skills/skillsSlice"
+import { StyledInput } from "../elements/Input"
+import { ReactComponent as UpdateIcon } from "../assets/icons/update-icon.svg"
+import { ReactComponent as EditIcon } from "../assets/icons/edit-icon.svg"
+import { ReactComponent as RemoveIcon } from "../assets/icons/remove-icon.svg"
+
+function Skill({ skill }) {
+  const dispatch = useDispatch()
+  const [inputValue, setInputValue] = useState()
```
Comment (ArturHorbenko) [inline src/layout/Skill.js]: Let's have more specific naming to the state for skill name, so if we'll have more inputs in this component -> we won't need to rename this one.

File: src/pages/skills/AddSkill.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    const isUser = localStorage.getItem("user_token")
+    if (!isUser) {
+      navigate("/auth/sign-in")
+    }
+    dispatch(getSkills())
+    return () => {
+      dispatch(reset())
+    }
+  }, [dispatch])
+  const { skills } = useSelector((state) => state.skills)
```
Comment (ArturHorbenko) [inline src/pages/skills/AddSkill.js]: Let's move skills state to the top. It's the most important data in this component, would be good to have skills state visible right away.

File: src/pages/skills/AddSkill.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            onChange={handleAddNewSkill}
+            id="project_stack"
+            placeholder="Add new skill"
+            name="skill"
+            // error={errors.project_stack}
+          />
+          <Button
+            onClick={(event) => onSubmitNewSkill(event)}
+            type="submit"
+            className="w-auto mt-[30px]"
```
Comment (ArturHorbenko) [inline src/pages/skills/AddSkill.js]: Let's make the button disabled if input is empty, so user won't be able to submit without any data.

File: src/features/skills/skillsSlice.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    }
+  }
+)
+
+// DELETE remove skill
+export const deleteSkill = createAsyncThunk(
+  "deleteSkill",
+  async (skillId, thunkAPI) => {
+    try {
+      return await instance.delete(`/skills/${skillId}`)
```
Comment (ArturHorbenko) [inline src/features/skills/skillsSlice.js]: Better concat the constant for current route ( `CURRENT_SLICE_ROUTE`) with `skillId`, same for `putSkill` request.

**Review by ArturHorbenko — APPROVED** [review]
Hi @hys25, 
You did a great job and decent progress making the whole CRUD from scratch. I left some comments about naming and small corrections.

**Comment by ArturHorbenko** [comment]
LGTM

**Pull request merged**